utils/kubernetes/describe/describe_test.go

```diff
@@ -0,0 +1,88 @@
+package describe
+
+import (
+	"testing"
+
+	meshkitkube "github.com/layer5io/meshkit/utils/kubernetes"
+)
+
+/*
+  The MockDescriber is used in the implementation of the Describe function to create a mock output that is returned when the
+  function is called with specific options
+  it takes in a Client object and a DescriberOptions object, and returns a string and an error
+*/
+type MockDescriber struct {
+	DescribeFunc func(*meshkitkube.Client, DescriberOptions) (string, error)
+}
+
+// Describe method of the MockDescriber simply calls DescribeFunc and returns the result.
+func (m *MockDescriber) Describe(client *meshkitkube.Client, options DescriberOptions) (string, error) {
+	return m.DescribeFunc(client, options)
+}
+
+func TestDescribe(t *testing.T) {
+	//meshkitkube.Client  provides the ability to interact with the Kubernetes API server
+
+	//set up mock client client to return expected responses
+	mockClient := meshkitkube.Client{}
+
+	//create test cases
+	testCases := []struct {
+		Name           string
+		Options        DescriberOptions
+		DescribeFunc   func(*meshkitkube.Client, DescriberOptions) (string, error)
+		ExpectedOutput string
+		ExpectedError  error
+	}{
+		{
+			Name: "describe pod",
+			Options: DescriberOptions{
+				Name:      "test-pod",
+				Namespace: "test-namespace",
+				Type:      Pod,
+			},
+			/*
+				DescribeFunc field is a function that takes in a Client
+				object and a DescriberOptions object, and returns a string and an error
+			*/
+			DescribeFunc: func(client *meshkitkube.Client, options DescriberOptions) (string, error) {
+				return "Name: test-pod\nNamespace: test-namespace\n", nil
+			},
+			ExpectedOutput: "Name: test-pod\nNamespace: test-namespace\n",
+			ExpectedError:  nil,
+		},
+		{
+			Name: "describe deployment",
+			Options: DescriberOptions{
+				Name:      "test-deployment",
+				Namespace: "test-namespace",
+				Type:      Deployment,
+			},
+			DescribeFunc: func(client *meshkitkube.Client, options DescriberOptions) (string, error) {
+				return "Name: test-deployment\nNamespace: test-namespace\n", nil
+			},
+			ExpectedOutput: "Name: test-deployment\nNamespace: test-namespace\n",
+			ExpectedError:  nil,
+		},
+	}
+
+	//run test cases
+	for _, tc := range testCases {
```

Comment:
Thanks for your contribution. I beleive we really need unit test for all the repos. All the code is good and clean but there is consideration about using for run all the test cases manually.
Please check code below, we prefer use this way.

https://github.com/meshery/meshery-operator/blob/1ace1d307c2278325b3c7b9c741f7dc7af912529/api/v1alpha1/meshsync_types_test.go#L62-L95

Reply:
ok this method seems new to me tho,  but i would see how i can implement it . I would reach out incase i have issues or i am stuck.